storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncSupervisorEvent.java

```diff
@@ -0,0 +1,626 @@
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SyncSupervisorEvent implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SyncSupervisorEvent.class);
+
+    private EventManager syncSupEventManager;
+    private EventManager syncProcessManager;
+    private IStormClusterState stormClusterState;
+    private LocalState localState;
+    private SyncProcessEvent syncProcesses;
+    private SupervisorData supervisorData;
+
+    public SyncSupervisorEvent(SupervisorData supervisorData, SyncProcessEvent syncProcesses, EventManager syncSupEventManager,
+            EventManager syncProcessManager) {
+
+        this.syncProcesses = syncProcesses;
+        this.syncSupEventManager = syncSupEventManager;
+        this.syncProcessManager = syncProcessManager;
+        this.stormClusterState = supervisorData.getStormClusterState();
+        this.localState = supervisorData.getLocalState();
+        this.supervisorData = supervisorData;
+    }
+
+    @Override
+    public void run() {
+        try {
+            Map conf = supervisorData.getConf();
+            Runnable syncCallback = new EventManagerPushCallback(this, syncSupEventManager);
+            List<String> stormIds = stormClusterState.assignments(syncCallback);
+            Map<String, Map<String, Object>> assignmentsSnapshot =
+                    getAssignmentsSnapshot(stormClusterState, stormIds, supervisorData.getAssignmentVersions().get(), syncCallback);
+            Map<String, List<ProfileRequest>> stormIdToProfilerActions = getProfileActions(stormClusterState, stormIds);
+
+            Set<String> allDownloadedTopologyIds = SupervisorUtils.readDownLoadedStormIds(conf);
+            Map<String, String> stormcodeMap = readStormCodeLocations(assignmentsSnapshot);
+            Map<Integer, LocalAssignment> existingAssignment = localState.getLocalAssignmentsMap();
+            if (existingAssignment == null) {
+                existingAssignment = new HashMap<>();
+            }
+
+            Map<Integer, LocalAssignment> allAssignment =
+                    readAssignments(assignmentsSnapshot, existingAssignment, supervisorData.getAssignmentId(), supervisorData.getSyncRetry());
+
+            Map<Integer, LocalAssignment> newAssignment = new HashMap<>();
+            Set<String> assignedStormIds = new HashSet<>();
+
+            for (Map.Entry<Integer, LocalAssignment> entry : allAssignment.entrySet()) {
+                if (supervisorData.getiSupervisor().confirmAssigned(entry.getKey())) {
+                    newAssignment.put(entry.getKey(), entry.getValue());
+                    assignedStormIds.add(entry.getValue().get_topology_id());
+                }
+            }
+
+            Set<String> srashStormIds = verifyDownloadedFiles(conf, supervisorData.getLocalizer(), assignedStormIds, allDownloadedTopologyIds);
```

Comment:
I'm not sure what `srachStormIds` means. It looks like it is storm ids that are in a bad state, so possibly `crashedStormIds` is what you were going for?